The sh4 and sparc targets have assembly implementations of clone.S that don't
get used, and thus the build breaks.  Also, sparc is missing a header file in
pthreads.old that exists in pthreads.new.

diff --git a/libc/sysdeps/linux/sh/Makefile.arch b/libc/sysdeps/linux/sh/Makefile.arch
index 92e262b..6cbc681 100644
--- a/libc/sysdeps/linux/sh/Makefile.arch
+++ b/libc/sysdeps/linux/sh/Makefile.arch
@@ -9,4 +9,4 @@
 CSRC := \
 	mmap.c pipe.c __init_brk.c brk.c sbrk.c pread_write.c cacheflush.c
 
-SSRC := setjmp.S __longjmp.S ___fpscr_values.S vfork.S
+SSRC := setjmp.S __longjmp.S ___fpscr_values.S vfork.S clone.S
diff --git a/libc/sysdeps/linux/sparc/Makefile.arch b/libc/sysdeps/linux/sparc/Makefile.arch
index d0cae9f..820b2fa 100644
--- a/libc/sysdeps/linux/sparc/Makefile.arch
+++ b/libc/sysdeps/linux/sparc/Makefile.arch
@@ -13,7 +13,7 @@ SSRC := \
 
 ifneq ($(UCLIBC_HAS_THREADS_NATIVE),y)
 CSRC += sigaction.c
-SSRC += fork.S vfork.S
+SSRC += fork.S vfork.S clone.S
 endif
 
 # check weather __LONG_DOUBLE_128__ is defined (long double support)
--- /dev/null	2011-07-15 06:34:19.820919508 -0500
+++ uClibc/libpthread/linuxthreads.old/sysdeps/sparc/tcb-offsets.h	2011-07-23 19:53:02.079880519 -0500
@@ -0,0 +1 @@
+#include "../../../linuxthreads/sysdeps/pthread/tcb-offsets.h"
